tools/mtmd/mtmd.cpp

```diff
@@ -580,6 +591,69 @@ struct decode_embd_batch {
     }
 };
 
+// Helper function for decoding an image whose embeddings have already been calculated
+int32_t mtmd_helper_decode_image(
+        mtmd_context * ctx,
+        struct llama_context * lctx,
+        const mtmd_image_tokens * image_tokens,
```

Comment:
```suggestion
        const mtmd_input_chunk * chunk,
```

Unless I'm missing something, I think here we can simplify accept a chunk, right? The `mtmd_image_tokens` are always managed by the chunk, so if you copy a chunk, the tokens will also be copied. This way, we can avoid exposing `mtmd_image_tokens_copy/free` to the public API.

Also I'm not very confident about exposing `mtmd_image_tokens_copy/free` to the public, as there is a chance the `mtmd_image_tokens` API will be changed in a near future. To make it less messy, I want it to be always attached to a chunk.

Reply:
Totally hear this, agree and seems doable with a minor check to ensure that the chunk actually has `mtmd_image_tokens`? I guess this was the think that made me feel that passing any higher-level container would be excessive, but definitely see how the API surface increase is not desirable

Reply:
yes we can have a check to ensure if the chunk is an image chunk (by checking the type). If it is not the correct type, we return with a specific code, maybe code `-1`

Reply:
done